packages/expo-cli/src/commands/eas-build/build/utils/configureUpdatesAsync.ts

```diff
@@ -0,0 +1,276 @@
+import { Platform } from '@expo/build-tools';
+import { getConfig, getExpoSDKVersion, getPackageJson } from '@expo/config';
+import { UserManager } from '@expo/xdl';
+import chalk from 'chalk';
+import dedent from 'dedent';
+import figures from 'figures';
+import * as fs from 'fs-extra';
+import glob from 'glob';
+import { reject } from 'lodash';
+import ora from 'ora';
+import path from 'path';
+import xcode from 'xcode';
+import { DOMParser, XMLSerializer } from 'xmldom';
+
+import { gitAddAsync } from '../../../../git';
+import log from '../../../../log';
+import * as gitUtils from '../../utils/git';
+
+interface UpdateOptions {
+  sdkVersion: string;
+  runtimeVersion?: string;
+  updateUrl: string;
+}
+
+const platformDisplayNames = {
+  [Platform.Android]: 'Android',
+  [Platform.iOS]: 'iOS',
+};
+
+export default async function configureUpdatesAsync({
+  projectDir,
+  nonInteractive,
+  platform,
+}: {
+  projectDir: string;
+  nonInteractive: boolean;
+  platform: Platform;
+}) {
+  const user = await UserManager.ensureLoggedInAsync();
+
+  const packageJson = getPackageJson(projectDir);
+
+  if (!(packageJson.dependencies && packageJson.dependencies['expo-updates'])) {
+    return;
+  }
+
+  const spinner = ora(`Configuring expo-updates for ${platformDisplayNames[platform]}`);
+
+  const { exp } = getConfig(projectDir);
+  const sdkVersion = getExpoSDKVersion(projectDir, exp);
+
+  const options = {
+    sdkVersion,
+    runtimeVersion: exp.runtimeVersion,
+    updateUrl: `https://exp.host/@${user.username}/${exp.slug}`,
+  };
+
+  switch (platform) {
+    case Platform.Android:
+      await configureUpdatesAndroid(projectDir, options);
+      break;
+    case Platform.iOS:
+      await configureUpdatesIOS(projectDir, options);
+      break;
+  }
+
+  try {
+    await gitUtils.ensureGitStatusIsCleanAsync();
+    spinner.succeed();
+  } catch (err) {
+    if (err instanceof gitUtils.DirtyGitTreeError) {
+      spinner.succeed(
+        `We configured expo-updates in your project for ${platformDisplayNames[platform]}`
+      );
+      log.newLine();
+
+      try {
+        await gitUtils.reviewAndCommitChangesAsync(
+          `Configure expo-updates for ${platformDisplayNames[platform]}`,
+          { nonInteractive }
+        );
+
+        log(`${chalk.green(figures.tick)} Successfully committed the configuration changes.`);
+      } catch (e) {
+        throw new Error(
+          "Aborting, run the command again once you're ready. Make sure to commit any changes you've made."
+        );
+      }
+    } else {
+      spinner.fail();
+      throw err;
+    }
+  }
+}
+
+async function configureUpdatesIOS(
```

Comment:
We have a log of similar functionality in the expo/config package already, maybe it's better to have this live there for now.

Reply:
agree with this. i think we should also use the `xml2js` package as we do in `@expo/config` rather than `xmldom`

Reply:
@brentvatne the reason I use `xmldom` is because it preserves whitespace and comments, so the formatting of android manifest is reasonably preserved.

Reply:
Formatting logic is done manually in here https://github.com/expo/expo-cli/blob/2c51e2f7b549ef6e2b3cb86f721959ad01ae6c5d/packages/config/src/android/Manifest.ts#L22

Reply:
@EvanBacon yeah, but xmldom preserves the original whitespace and comments. I've removed it for now since I have moved the logic to `@expo/config`, but it's something to keep in mind.